feat(event): add event tagging for categorization

M-e-d-A opened the pull request:
# Description:
## Purpose of the change
This PR introduces event tagging for categorization and filtering, as defined in sub-issue #409.

It includes:
* `EventTag` enum with common event categories and formats
* `tags` field added to the `Event` model
* Tag selection UI in event creation and editing forms
* Support for selecting up to 5 tags per event
* Unit and UI tests for the new functionality

## Related issues
Closes #409
Part of #406

## How to test
1. Check out the branch.
2. Run the tests in `EventTest.kt`, `EditEventFormViewModelTest.kt`, `CreateEventFormComposeTest.kt`, and `EditEventFormComposeTest.kt`.
3. Verify all tests pass.
4. Build and run the app, create or edit an event, and verify tag selection works and is saved.
5. Verify the 5-tag limit is enforced.

**Note:** 
- An LLM was used for the sole and exclusive purpose of improving the clarity and formatting of the PR description.
- This is PR 3 of 4 in the Like & Recommender implementation series.

**Related PRs in this series:**
- #411
- #412
- #413 *(this PR)*
- #414

<table>
  <tr>
    <td align="center">
      <img src="https://github.com/user-attachments/assets/51807568-a39d-4aa8-a2ce-b52e124aa302" alt="Tags choice upon creation (five selected)" width="300px">
      <br>
      <em>Tags choice upon creation (five selected)</em>
    </td>
    <td align="center">
      <img src="https://github.com/user-attachments/assets/d5517313-4f63-41b8-980a-b356ade59f25" alt="Tags choice upon creation (none selected)" width="300px">
      <br>
      <em>Tags choice upon creation (none selected)</em>
    </td>
  </tr>
</table>

Bouazz-a reviewed (approved):
**Overall Assessment**
Solid event tagging implementation with clean enum design, proper UI component usage, and good test coverage. The 5-tag limit enforcement and multi-category grouping make this feature production-ready. Minor polish items suggested as nitpicks.

**Strengths**
- Clean enum architecture: `EventTag` with display values, `fromString()` parsing, and logical categorization (Theme/Format/Setting) makes this extensible and maintainable.
- Proper Compose usage: `FilterChip` with `FlowRow` is the right component choice; state management via ViewModel keeps UI logic separate.
- 5-tag limit correctly enforced: Logic in `toggleTag()` prevents exceeding limit while allowing removal at any time.
- Well-organized UI: Grouping tags by category ("Theme", "Format", "Setting & Cost") improves discoverability compared to a flat list.

**Nitpicks**
- Consider showing tag counter in UI (`3/5 selected`) so users know when they're approaching the limit.
- Extract magic number `5` to `const val MAX_EVENT_TAGS = 5` for maintainability.
- Add accessibility semantics to `FilterChip` for screen reader support (`contentDescription`).
- Consider stable IDs for enum serialization to prevent data breakage if you rename enum values later.
- Move tag groups to `EventTag.getGroups()` companion function instead of hardcoding in composable.

Otherwise, looks good to me!

M-e-d-A commented:
>     * Extract magic number `5` to `const val MAX_EVENT_TAGS = 5` for maintainability.
>     * Move tag groups to `EventTag.getGroups()` companion function instead of hardcoding in composable.

Agreed. Moving the hardcoded grouping logic from the UI (EventFormFields) to the Domain model (Event.kt) adheres better to Separation of Concerns. We will also define the maximum tag count constant in the ViewModel (where the logic resides) to avoid magic numbers.

>     * Consider showing tag counter in UI (`3/5 selected`) so users know when they're approaching the limit.
>     * Add accessibility semantics to `FilterChip` for screen reader support (`contentDescription`).

Updated the UI to show the dynamic counter using the new constant, and ensured the list is iterated from the domain model. Also added contentDescription for better accessibility.
